docs/tutorials/deploy_openfaas_workload.md

```diff
@@ -0,0 +1,242 @@
+2. `handler`: handler is the entry point of your function, the type is string, this field is optional, and also could read from parameters.
+3. `environment`: the type of environment is array, and you should write the environment list in the field value. Each environment could be read from parameters.
+
+## Create Application Configuration to use the Component
+
+After create the component, we should create an application configuration to use it.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: ApplicationConfiguration
+metadata:
+  name: openfaas
+spec:
+  components:
+    - componentName: openfaas
+      instanceName: gofast
+      parameterValues:
+        - name: image
+          value: functions/nodeinfo
+        - name: handler
+          value: "node main.js"
+        - name: write_debug
+          value: "false"
+```
+
+Yes, just set the `parameterValues` to override parameters in component.
+
+We have write these two example yaml files in [openfaasapp.yaml](../../examples/openfaasapp.yaml).
+
+We could deploy it by:
+
+```shell script
+$ kubectl apply -f examples/openfaasapp.yaml
+componentschematic.core.oam.dev/openfaas configured
+applicationconfiguration.core.oam.dev/openfaas created
+``` 
+
+Then you could find functions have been created:
+
+```shell script
+$ kubectl get functions
+NAME       AGE
+nodeinfo   27s
+```
+
+You could also visit the function deployed:
+
+```shell script
+$ curl http://127.0.0.1:31112/function/nodeinfo
+Hostname: nodeinfo-86b679f76f-sgkbb
+
+Platform: linux
+Arch: x64
+CPU count: 2
+Uptime: 796748
+```
+
+## Apply trait
+
+Manual Scaler trait could also be applied to OpenFaaS workload. Just add this trait like what to do to core workload types like below:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: ApplicationConfiguration
+metadata:
+  name: openfaas
+spec:
+  components:
+    - componentName: openfaas
+      instanceName: nodeinfo
+      parameterValues:
+        - name: image
+          value: functions/nodeinfo
+        - name: handler
+          value: "node main.js"
+        - name: write_debug
+          value: "false"
+      traits:
+        - name: manual-scaler
+          parameterValues:
+            - name: replicaCount
+              value: 2
+```
+
+After change the Application Configuration yaml file, apply it.
+
+```
+$ kubeclt apply -f examples/openfaasapp.yaml
+componentschematic.core.oam.dev/openfaas unchanged
+applicationconfiguration.core.oam.dev/openfaas configured
+```
+
+Then we could see the replica of the function has changed.
+
+```shell script
+$ kubectl get functions nodeinfo -o yaml
+apiVersion: openfaas.com/v1alpha2
+kind: Function
+metadata:
+  name: nodeinfo
+  namespace: default
+spec:
+  environment:
+    write_debug: "false"
+  handler: node main.js
+  image: functions/nodeinfo
+  name: nodeinfo
+  replicas: 2
+```
+
+If you want to change the replicas, just change the application configuration yaml and apply it again.
```

Comment:
Should continue to add an example about how to scale. Don't be lazy :-)

Reply:
In real world practices, the OpenFaaS `Component` is expected to be defined by Devs, while `ApplicationConfiguration` and manual-scaler `Trait` be managed by App Ops. You could check further practices guideline in [roles and responsibilities](https://github.com/oam-dev/spec/blob/master/2.overview_and_terminology.md#roles-and-responsibilities) explanation of OAM spec.

Reply:
Cool, I will add it